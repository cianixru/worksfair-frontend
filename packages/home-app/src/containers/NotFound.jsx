import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import { Image, Transformation } from 'cloudinary-react';

import Footer from '../components/Footer/Footer';

class NotFound extends Component {
  render() {
    const compass = 'https://res.cloudinary.com/worksfair/image/upload/v1564834329/dev/qvwvfuxg6040ppipkanj.png';
    return (
      <Fragment>
        <Helmet>
          <title>Not Found - Worksfair</title>
        </Helmet>
        <div
          className="hero is-large has-background-light"
        >
          <div className="hero-body">
            <div className="container columns">
              <div className="column is-1"/>
              <div className="column is-7">
                <h1 className="has-text-extra-large">404</h1>
                <p className="has-text-large">We are sorry about this.</p>
                <p className="is-size-5">
                    The page you are looking for is not here.
                </p>
                <p className="margin-top-25">
                  <Link
                    to="/"
                    className="button is-info is-medium">
                    Go Home
                  </Link>
                </p>
              </div>
              <div className="column is-3 has-text-extra-large">
                <Image
                  cloudName="worksfair"
                  publicId={compass}
                  type="fetch"
                  className="offering-image">
                  <Transformation width="700" fetchFormat="auto" />
                </Image>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

NotFound.propTypes = {
  location: PropTypes.object,
};

export default NotFound;
