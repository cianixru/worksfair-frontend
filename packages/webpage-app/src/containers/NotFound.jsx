import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import dotenv from 'dotenv';
import { Image, Transformation } from 'cloudinary-react';

import Footer from '../components/Footer/Footer';

dotenv.config();

class NotFound extends Component {
  render() {
    const compass = 'https://res.cloudinary.com/worksfair/image/upload/v1564834329/dev/qvwvfuxg6040ppipkanj.png';
    return (
      <Fragment>
        <Helmet>
          <title>Not Found - Worksfair</title>
        </Helmet>
        <div
          className="hero is-large has-background-light"
        >
          <div className="hero-body">
            <div className="container columns">
              <div className="column is-1"/>
              <div className="column is-7">
                <h1 className="has-text-extra-large">404</h1>
                <p className="has-text-large">We are sorry about this.</p>
                <p className="is-size-5">
                    The page you are looking for is not here.
                </p>
                <p className="margin-top-25">
                  <a
                    href={process.env.REACT_APP_URL}
                    className="button is-info is-medium">
                    Go To Worksfair
                  </a>
                </p>
              </div>
              <div className="column is-3 has-text-extra-large">
                <Image
                  cloudName="worksfair"
                  publicId={compass}
                  type="fetch"
                  className="offering-image">
                  <Transformation width="700" fetchFormat="auto" />
                </Image>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

NotFound.propTypes = {
  location: PropTypes.object,
};

export default NotFound;
