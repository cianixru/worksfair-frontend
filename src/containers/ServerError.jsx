import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import Footer from '../components/Footer/Footer';
import compass from '../assets/404-compass.png';

class ServerError extends Component {
  render() {
    return (
      <Fragment>
        <Helmet>
          <title>Server Error - Worksfair</title>
        </Helmet>
        <div
          className="hero is-large has-background-light"
        >
          <div className="hero-body">
            <div className="container columns">
              <div className="column is-1"/>
              <div className="column is-7">
                <h1 className="has-text-extra-large">500</h1>
                <p className="has-text-large">We are sorry about this.</p>
                <p className="is-size-5">
                  An error occurred on our server, please bear with us.
                </p>
                <p className="margin-top-25">
                  <Link
                    to="/"
                    className="button is-info is-medium">
                    Go Home
                  </Link>
                </p>
              </div>
              <div className="column is-3 has-text-extra-large">
                <img src={compass} alt="404-compass" />
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

ServerError.propTypes = {
  location: PropTypes.object,
};

export default ServerError;
