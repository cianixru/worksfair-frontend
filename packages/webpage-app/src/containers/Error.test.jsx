import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import ServerError from './ServerError';
import NotFound from './NotFound';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { newWebpage: {} }
});
describe('NotFound.jsx', () => {
    test('Should display the NotFound Component', () => {
        const component = render(
        <Provider store={mockStore}>
            <Router>
              <NotFound />
            </Router>
        </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });
});

describe('ServerError.jsx', () => {
  test('Should display the ServerError Component', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <ServerError />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
