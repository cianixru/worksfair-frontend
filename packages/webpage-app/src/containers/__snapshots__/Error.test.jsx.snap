// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.jsx Should display the NotFound Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hero is-large has-background-light"
      >
        <div
          class="hero-body"
        >
          <div
            class="container columns"
          >
            <div
              class="column is-1"
            />
            <div
              class="column is-7"
            >
              <h1
                class="has-text-extra-large"
              >
                404
              </h1>
              <p
                class="has-text-large"
              >
                We are sorry about this.
              </p>
              <p
                class="is-size-5"
              >
                The page you are looking for is not here.
              </p>
              <p
                class="margin-top-25"
              >
                <a
                  class="button is-info is-medium"
                  href="http://localhost:3001"
                >
                  Go To Worksfair
                </a>
              </p>
            </div>
            <div
              class="column is-3 has-text-extra-large"
            >
              <img
                class="offering-image"
                src="http://res.cloudinary.com/worksfair/image/fetch/f_auto/https://res.cloudinary.com/worksfair/image/upload/v1564834329/dev/qvwvfuxg6040ppipkanj.png"
              />
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer "
      >
        <div
          class="content has-text-centered"
        >
          Copyright 
          2019
           Worksfair. All Rights Reserved.
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="hero is-large has-background-light"
    >
      <div
        class="hero-body"
      >
        <div
          class="container columns"
        >
          <div
            class="column is-1"
          />
          <div
            class="column is-7"
          >
            <h1
              class="has-text-extra-large"
            >
              404
            </h1>
            <p
              class="has-text-large"
            >
              We are sorry about this.
            </p>
            <p
              class="is-size-5"
            >
              The page you are looking for is not here.
            </p>
            <p
              class="margin-top-25"
            >
              <a
                class="button is-info is-medium"
                href="http://localhost:3001"
              >
                Go To Worksfair
              </a>
            </p>
          </div>
          <div
            class="column is-3 has-text-extra-large"
          >
            <img
              class="offering-image"
              src="http://res.cloudinary.com/worksfair/image/fetch/f_auto/https://res.cloudinary.com/worksfair/image/upload/v1564834329/dev/qvwvfuxg6040ppipkanj.png"
            />
          </div>
        </div>
      </div>
    </div>
    <footer
      class="footer "
    >
      <div
        class="content has-text-centered"
      >
        Copyright 
        2019
         Worksfair. All Rights Reserved.
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ServerError.jsx Should display the ServerError Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hero is-large has-background-light"
      >
        <div
          class="hero-body"
        >
          <div
            class="container columns"
          >
            <div
              class="column is-1"
            />
            <div
              class="column is-7"
            >
              <h1
                class="has-text-extra-large"
              >
                500
              </h1>
              <p
                class="has-text-large"
              >
                We are sorry about this.
              </p>
              <p
                class="is-size-5"
              >
                An error occurred on our server, please bear with us.
              </p>
              <p
                class="margin-top-25"
              >
                <a
                  class="button is-info is-medium"
                  href="/"
                >
                  Go Home
                </a>
              </p>
            </div>
            <div
              class="column is-3 has-text-extra-large"
            >
              <img
                alt="404-compass"
                src="404-compass.png"
              />
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer "
      >
        <div
          class="content has-text-centered"
        >
          Copyright 
          2019
           Worksfair. All Rights Reserved.
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="hero is-large has-background-light"
    >
      <div
        class="hero-body"
      >
        <div
          class="container columns"
        >
          <div
            class="column is-1"
          />
          <div
            class="column is-7"
          >
            <h1
              class="has-text-extra-large"
            >
              500
            </h1>
            <p
              class="has-text-large"
            >
              We are sorry about this.
            </p>
            <p
              class="is-size-5"
            >
              An error occurred on our server, please bear with us.
            </p>
            <p
              class="margin-top-25"
            >
              <a
                class="button is-info is-medium"
                href="/"
              >
                Go Home
              </a>
            </p>
          </div>
          <div
            class="column is-3 has-text-extra-large"
          >
            <img
              alt="404-compass"
              src="404-compass.png"
            />
          </div>
        </div>
      </div>
    </div>
    <footer
      class="footer "
    >
      <div
        class="content has-text-centered"
      >
        Copyright 
        2019
         Worksfair. All Rights Reserved.
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
